tests/test_spark.py

```diff
@@ -430,3 +433,41 @@ def test_task_stdout():
         assert stdout_file["size"] > 0, "stdout file should have content"
     finally:
         sdk_install.uninstall(utils.SPARK_PACKAGE_NAME, service_name)
+
+
+@pytest.mark.sanity
+def test_handling_wrong_request_to_spark_dispatcher():
+    #submit a correct job
+    service_name = utils.FOLDERED_SPARK_SERVICE_NAME
+    utils.require_spark(service_name=service_name)
+    results  = sdk_cmd.run_raw_cli("dcos spark run --submit-args=\"--class org.apache.spark.examples.SparkPi https://downloads.mesosphere.com/spark/assets/spark-examples_2.11-2.0.1.jar 30\"")
+
+    #get it's submission id
+    submission_id = results.stdout.split(" ")[-1]
+
+    #submit a job through the REST interface with incorrect appArgs
+    sdk_cmd.run_raw_cli("node ssh --master-proxy --leader")
+    results = subprocess.run("hostname", shell=True, universal_newlines=True, check=True)
+    host = results.stdout
+    headers = {
+        'Content-Type': 'application/json;charset=UTF-8',
+    }
+    data = {"appResource": "https://downloads.mesosphere.com/spark/assets/spark-examples_2.11-2.0.1.jar",
```

Comment:
we don't need to run actual Spark Job, what we need to check is that Dispatcher accepts valid JSON request and _attempts to start_ a job, we are testing Dispatcher liveness in case of JSON request with missing fields.

Valid request:
```
{
    "action": "CreateSubmissionRequest",
    "clientSparkVersion": "1.2.3",
    "appResource": "noop-test.jar",
    "mainClass": "noop.Test",
    "appArgs": ["arg1", "arg2"],
    "environmentVariables": {
        "PATH": "/dev/null"
    },
    "sparkProperties": {
        "spark.app.name": "TestDispatcher"
    }
}
```

Invalid request (no *appArgs* field):
```
{
    "action": "CreateSubmissionRequest",
    "clientSparkVersion": "1.2.3",
    "appResource": "noop-test.jar",
    "mainClass": "noop.Test",
    "environmentVariables": {
        "PATH": "/dev/null"
    },
    "sparkProperties": {
        "spark.app.name": "TestDispatcher"
    }
}
```

Test ahould assert that after submitting an invalid request, the Dispatcher under test is able to accept valid request and _attempts to start a job_

Reply:
Passed the two requests as Python requests object.